sdk/core/azure_core_test/src/lib.rs

```diff
@@ -70,6 +84,14 @@ fn find_ancestor(dir: impl AsRef<Path>, name: &str) -> azure_core::Result<PathBu
         if path.exists() {
             return Ok(path);
         }
+
+        let path = dir.join(".git");
+        if path.exists() {
```

Comment:
The error message appears to be inconsistent with the test - if there's a <path>.git file, you report that "<name> not found under <path>".

I'm sure there's a reason for this, but a comment might be helpful to understand the scenario that is being checked here.

Reply:
Thank you. I totally got my wires cross on that one.

Reply:
Per offline discussion, I'll comment below as to the purpose and clarify the error message a bit as well.